pkg/kapis/logging/v1alpha2/handler.go

```diff
@@ -1 +1,190 @@
 package v1alpha2
+
+import (
+	"github.com/emicklei/go-restful"
+	"kubesphere.io/kubesphere/pkg/api"
+	"kubesphere.io/kubesphere/pkg/models/logging"
+	"kubesphere.io/kubesphere/pkg/simple/client/k8s"
+	loggingclient "kubesphere.io/kubesphere/pkg/simple/client/logging"
+	util "kubesphere.io/kubesphere/pkg/utils/stringutils"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	LevelCluster = iota
+	LevelContainer
+
+	// query type, default to `query`
+	TypeStat   = "statistics"
+	TypeHist   = "histogram"
+	TypeExport = "export"
+)
+
+type handler struct {
+	k8s.Client
+	logging.LoggingOperator
+}
+
+func newHandler(k k8s.Client, l loggingclient.Interface) *handler {
+	return &handler{k, logging.NewLoggingOperator(l)}
+}
+
+func (h handler) handleClusterQuery(req *restful.Request, resp *restful.Response) {
+	Get(req, LevelCluster, h.Client, h.LoggingOperator, resp)
+}
+
+func (h handler) handleContainerQuery(req *restful.Request, resp *restful.Response) {
+	Get(req, LevelContainer, h.Client, h.LoggingOperator, resp)
+}
+
+func Get(req *restful.Request, lvl int, client k8s.Client, loggingOperator logging.LoggingOperator, resp *restful.Response) {
```

Comment:
If `Get` only called inside handler, then it's unnecessary to make `Get` an exported function. Also I think it's better to define `Get` a member method of handler

Reply:
the reason is that tenant api depends this, i have suggested @wansir moving codes for multi-tenant logging to logging module